test/calltoaction.spec.js

```diff
@@ -1,151 +1,133 @@
+            $fixture.parentElement.removeChild($fixture);
+          });
 
-      let callToAction;
+          it('is empty if the HTML element lacks a data-cookie-name-root attribute', function () {
+            const id = fixtures.generateRandomId();
+            const $fixture = fixtures.generateHTMLWithCookieDetails(fixtures.getCookieNameRoot(), id);
+            delete $fixture.dataset.cookieNameRoot;
 
-      beforeEach(() => {
-        expireCookie(expectedCookieName);
-        document.querySelector('.call-to-action-wrapper').classList.remove('call-to-action-wrapper--js-shown');
-      });
+            const callToAction = new CallToAction($fixture);
+            expect(callToAction.dismissible.cookieName).to.equal('');
 
-      it('has the CSS class "call-to-action-wrapper--js-shown"', () => {
-        callToAction = new CallToAction($elm);
-        expect($elm.classList.contains('call-to-action-wrapper--js-shown')).to.be.true;
-      });
+            $fixture.parentElement.removeChild($fixture);
+          });
 
-    });
+        });
 
-    describe('the cookie name', () => {
+        describe('expiry date', function () {
 
-      it('is the string "callToAction_" appended with the id attribute of the component\'s HTML element', () => {
-        expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.be.empty;
-        (new CallToAction($elm)).dismiss();
-        expect(utils.getCookieValue(expectedCookieName, document.cookie)).to.equal('true');
-      });
+          it('is configured with the value of data-cookie-expires HTML attribute, if set', function () {
+            const id = fixtures.generateRandomId();
+            const expiry = 'Tue, 19 January 2038 03:14:07 UTC';
+            const $fixture = fixtures.generateHTMLWithCookieDetails(fixtures.getCookieNameRoot(), id, expiry);
+            const callToAction = new CallToAction($fixture);
+            expect(callToAction.dismissible.cookieExpiryDate).to.equal(expiry);
 
-      context('when the component\'s HTML element is missing an id attribute', () => {
+            fixtures.clearCookie(id);
+            expect(utils.getCookieValue(`${fixtures.getCookieNameRoot()}${id}`, document.cookie), 'cookie shouldn\'t be set').to.equal('');
 
-        let $elmNoId;
-        let callToActionWithElmNoId;
+          });
 
-        beforeEach(() => {
-          $elmNoId = $elm.cloneNode(true);
-          $elmNoId.setAttribute('id', '');
-          $elm.parentElement.appendChild($elmNoId);
+          it('is configured with a day offset of the value of data-cookie-duration HTML attribute, if set', function () {
+            const id = fixtures.generateRandomId();
 
-          callToActionWithElmNoId = new CallToAction($elmNoId);
-        });
+            // 7 days in the future
+            let expiry = new Date(new Date());
+            expiry.setDate(expiry.getDate() + 7);
+            expiry = expiry.toUTCString();
```

Comment:
given the second precision, this may be vulnerable to race conditions where the tests expected values becomes slightly off if the test code and the production code run too far apart. Any way to ~mock~ stub `Date`?

Reply:
Removed the underlying feature of setting `cookieDuration`, and so also this test.